src/ocean/text/convert/Formatter.d

```diff
@@ -478,7 +478,30 @@ private void handle (T) (T v, FormatInfo f, scope FormatterSink sf, scope ElemSi
         nullWrapper(&v, se(v.toString(), f), se("null", f));
     else static if (is(T == interface))
         handle!(Object)(cast(Object) v, f, sf, se);
+    else static if (isInputRange!T)
+    {
+        bool started;
+        Flags old_flag = f.flags;
+        f.flags |= Flags.Nested;
+        while (!v.empty)
+        {
+            if (!started)
+            {
+                started = true;
+                sf("[ ");
+            }
+            else
+                sf(", ");
 
+            handle(v.front, f, sf, se);
+            v.popFront();
+        }
+        if (started)
+            sf(" ]");
+        else
+            sf("[]");
+        f.flags = old_flag;
+    }
```

Comment:
I think we can simplify this a little bit:
```suggestion
        if (v.empty)
        {
            sf("[]");
            return;
        }
        
        Flags old_flag = f.flags;
        f.flags |= Flags.Nested;

        sf("[ ");
        while (1)
        {
            handle(v.front, f, sf, se);
            v.popFront();
            if (v.empty)
                break;
            else
                sf(", ");
        }
        sf(" ]");
        f.flags = old_flag;
    }
```

But to make sure this works I suggest adding two additional test-cases:
- Test with a range with 1 element
- Test with an empty range.

Reply:
That's a good suggestion. They're all applied.